app/namespace.go

```diff
@@ -1425,6 +1435,66 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
 		},
 	})
 
+	// Note: the following commands are depend on previous commands. Please don't add more commands in the middle
+	exportCommand := subCommands[len(subCommands)-1]
+
+	// --- GCP Sink Related Commands ---
+	// Will move under export command once GCP sink is under public preview
+	if IsFeatureEnabled(GCPSinkFeatureFlag) {
+		exportCommand.Subcommands = append(exportCommand.Subcommands, &cli.Command{
+			Name:    "gcp-sink",
+			Usage:   "Manage GCP export sinks",
+			Aliases: []string{"gcp"},
+			Subcommands: []*cli.Command{
+				{
+					Name:    "validate",
+					Aliases: []string{"v"},
+					Usage:   "Validate gcp export sink",
+					Flags: []cli.Flag{
+						NamespaceFlag,
+						sinkNameFlag,
+						serviceAccountPrincipalFlag,
+						gcsBucketFlag,
+					},
+					Action: func(ctx *cli.Context) error {
+						namespace := ctx.String(NamespaceFlagName)
+						_, err := c.getNamespace(namespace)
+						if err != nil {
+							return fmt.Errorf("unable to get existing namespace: %v", err)
+						}
+
+						saName, projectName, err := parseSAPrincipal(ctx.String(serviceAccountPrincipalFlag.Name))
+						if err != nil {
+							return fmt.Errorf("validation failed: %v", err)
+						}
+
+						validateRequest := &namespaceservice.ValidateExportSinkRequest{
+							Namespace: namespace,
+							Spec: &sink.ExportSinkSpec{
+								Name:            ctx.String(sinkNameFlag.Name),
+								DestinationType: sink.EXPORT_DESTINATION_TYPE_GCS,
+								GcsSink: &sink.GCSSpec{
+									GcpProjectName: projectName,
+									BucketName:     ctx.String(gcsBucketFlag.Name),
+									SaName:         saName,
+								},
+							},
+						}
+
+						_, err = c.client.ValidateExportSink(c.ctx, validateRequest)
+
+						if err != nil {
+							return fmt.Errorf("validation failed with error %v", err)
+						}
+
+						fmt.Println("Validate test file can be written to the sink successfully")
```

Comment:
[Nit] re-phrase/re-write: "Temporal Cloud was able to write test data to the sink"